src/R/Components/Impl/ConnectionManager/Implementation/ViewModel/ConnectionManagerViewModel.cs

```diff
@@ -144,7 +147,27 @@ internal sealed class ConnectionManagerViewModel : BindableBase, IConnectionMana
             TryStartEditing(connection);
         }
 
-        public Task TestConnectionAsync(IConnectionViewModel connection) => Task.CompletedTask;
+        public async Task TestConnectionAsync(IConnectionViewModel connection) {
+            _shell.AssertIsOnMainThread();
+            connection.IsTestingConnection = true;
+            connection.IsTestConnectionSucceeded = false;
+
+            try {
+                await _connectionManager.TestConnectionAsync(connection);
+                connection.IsTestConnectionSucceeded = true;
+                connection.TestConnectionResult = "Connection test passed!";
+            } catch (ArgumentException) {
+                connection.TestConnectionResult = "Connection path is invalid.";
+            } catch (RHostDisconnectedException exception) {
+                connection.TestConnectionResult = "Connection test failed:{0}".FormatInvariant(exception.Message);
```

Comment:
It takes `exception.Message` as a format parameter. 

---
In reply to: [79271500](https://github.com/Microsoft/RTVS/pull/2342#discussion_r79271500) [](ancestors = 79271500)